### PR TITLE
Fix various policy reporting issues

### PR DESCRIPTION
**Pull request by hausdorff**
See commit messages for more information.

Fixes pulumi/pulumi-policy#25.
Fixes pulumi/pulumi-policy#31.

### CODE REVIEW

**Comment by hausdorff** [comment]
@pgavlin we might want to pass this information back to the step executor more explicitly as part of `GenerateSteps`. I did it this way because I'm reasonably confident we did it like this way back in the day, but we can change it if we want to.

File: pkg/resource/deploy/step_generator.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ type stepGenerator struct {
 	plan *Plan   // the plan to which this step generator belongs
 	opts Options // options for this step generator
 
+	hasPolicyViolation bool
```
Comment (pgavlin) [inline pkg/resource/deploy/step_generator.go]: Nit: comment

File: pkg/resource/deploy/plan_executor.go [hunk 2]
```diff
@@ -184,6 +187,10 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 			case <-ctx.Done():
 				logging.V(4).Infof("planExecutor.Execute(...): context finished: %v", ctx.Err())
 
+				if pe.stepGen.hasPolicyViolation {
```
Comment (pgavlin) [inline pkg/resource/deploy/plan_executor.go]: Let's move these out of this invocation and just do a single check below (near `pe.stepExec.Errored()`)

**Comment by hausdorff** [comment]
@pgavlin what else shoudl we do before shipping this?

File: pkg/resource/deploy/plan_executor.go [hunk 3]
```diff
@@ -194,6 +194,10 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	pe.stepExec.WaitForCompletion()
 	logging.V(4).Infof("planExecutor.Execute(...): step executor has completed")
 
+	if pe.stepGen.hasPolicyViolations {
```
Comment (pgavlin) [inline pkg/resource/deploy/plan_executor.go]: Sorry, I was suggesting moving this down to line 206:

```
	// Figure out if execution failed and why. Step generation and execution errors trump cancellation.
	if res != nil || pe.stepExec.Errored() || pe.stepGen.hasPolicyViolations {
```

Comment (hausdorff) [reply pkg/resource/deploy/plan_executor.go]: @pgavlin Done. I actually like _not_ emitting the "x failed" message, but understand the desire to keep it. Anything else?

**Pull request merged**